### PR TITLE
Create endpoint for getting/creating time entries.

### PR DESCRIPTION
**Pull request by lydiagu**

### CODE REVIEW

File: orchestra/tests/helpers/fixtures.py [hunk 1]
```diff
@@ -104,6 +104,12 @@ class Meta:
     snapshots = {}
 
 
+class TimeEntryFactory(factory.django.DjangoModelFactory):
```
Comment (jrbotros) [inline orchestra/tests/helpers/fixtures.py:107]: cool

File: orchestra/tests/helpers/fixtures.py [hunk 2]
```diff
@@ -360,13 +366,12 @@ def _setup_workers(test_case, workers):
...
                             first_name='test_first_{}'.format(user_id),
                             last_name='test_last_{}'.format(user_id),
-                            password='test_{}'.format(user_id),
                             email='test_user_{}@test.com'.format(user_id)))
         test_case.workers[user_id] = WorkerFactory(user=user)
         test_case.clients[user_id] = Client()
         test_case.clients[user_id].login(
             username='test_user_{}'.format(user_id),
-            password='test_{}'.format(user_id))
+            password='defaultpassword')
```
Comment (jrbotros) [inline orchestra/tests/helpers/fixtures.py:374]: `force_login` available in 1.9

File: orchestra/views.py [hunk 3]
```diff
@@ -190,6 +193,33 @@ def submit_task_assignment(request):
...
+            time_entry_data = json.loads(request.body.decode())
+            if 'task_id' not in time_entry_data:
+                raise BadRequest('Include task id in request data')
+            task_id = time_entry_data.pop('task_id')
+            return save_time_entry(worker, task_id, time_entry_data)
+    except Task.DoesNotExist:
+        raise BadRequest('No task for given id')
+    except TaskAssignment.DoesNotExist:
+        raise BadRequest('Worker is not assigned to this task id.')
+    except TaskStatusError as e:
```
Comment (jrbotros) [inline orchestra/views.py]: can add `serializers.ValidationError` to this line

File: orchestra/views.py [hunk 4]
```diff
@@ -190,6 +193,33 @@ def submit_task_assignment(request):
...
+@login_required
+def time_entries(request):
+    worker = Worker.objects.get(user=request.user)
+    try:
+        if request.method == 'GET':
+            return time_entries_for_worker(worker,
+                                           task_id=request.GET.get('task-id'))
+        elif request.method == 'POST':
+            time_entry_data = json.loads(request.body.decode())
+            if 'task_id' not in time_entry_data:
```
Comment (jrbotros) [inline orchestra/views.py:206]: maybe check for all expected params here

File: orchestra/tests/test_views.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        self.assertEqual(data['error'], 400)
+
+    def test_time_entries_get(self):
+        resp = self.request_client.get(self.url)
+        data = json.loads(resp.content.decode())
+        self.assertEqual(len(data), self.tasks.count())
+
+    @patch('orchestra.views.time_entries_for_worker')
+    def test_time_entries_get_task_id(self, mock_func):
+        mock_func.return_value = [self.time_entry_data]
```
Comment (jrbotros) [inline orchestra/tests/test_views.py:60]: add this as a kwarg to the patch decorator

File: orchestra/tests/test_views.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        self.request_client.get(self.url,
+                                data={'task-id': self.tasks[0].id})
+        mock_func.assert_called_with(self.worker,
+                                     task_id=str(self.tasks[0].id))
+
+    @patch('orchestra.views.time_entries_for_worker')
```
Comment (jrbotros) [inline orchestra/tests/test_views.py]: same as above

File: orchestra/tests/test_views.py [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    def test_time_entries_get_task_missing(self, mock_func):
+        mock_func.side_effect = Task.DoesNotExist
+        resp = self.request_client.get(self.url)
+        self._verify_missing_task(resp)
+
+    @patch('orchestra.views.time_entries_for_worker')
+    def test_time_entries_get_worker_not_assigned(self, mock_func):
+        pass
```
Comment (jrbotros) [inline orchestra/tests/test_views.py]: implement this

File: orchestra/tests/test_views.py [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+    @patch('orchestra.views.save_time_entry')
+    def test_time_entries_post(self, mock_func):
+        mock_func.return_value = self.time_entry_data
```
Comment (jrbotros) [inline orchestra/tests/test_views.py:80]: same as above

File: orchestra/tests/test_views.py [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+        # Remove task_id from dictionary so we can compare with mock_func
+        # call.
+        self.time_entry_data.pop('task_id')
+        mock_func.assert_called_with(self.worker, self.tasks[0].id,
+                                     self.time_entry_data)
+
+    @patch('orchestra.views.save_time_entry')
+    def test_time_entries_post_task_missing(self, mock_func):
+        mock_func.side_effect = Task.DoesNotExist
```
Comment (jrbotros) [inline orchestra/tests/test_views.py]: same as above

File: orchestra/tests/test_views.py [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+        # Set task id in request body.
+        self.time_entry_data['task_id'] = self.tasks[0].id
+        resp = self.request_client.post(self.url,
+                                        data=json.dumps(self.time_entry_data),
+                                        content_type='application/json')
+        self._verify_missing_task(resp)
+
+    @patch('orchestra.views.save_time_entry')
+    def test_time_entries_post_worker_not_assigned(self, mock_func):
+        mock_func.side_effect = TaskAssignment.DoesNotExist
```
Comment (jrbotros) [inline orchestra/tests/test_views.py]: same as above

**Comment by jrbotros** [comment]
done reviewing

File: orchestra/views.py (hunk 4 above)
Comment (lydiagu) [inline orchestra/views.py:206]: that's handled by the serializer. It's a bit of a hack - task_id really should be a get parameter. but i stick it in the body

**Pull request merged**